### PR TITLE
Distance modification process optimization

### PR DESCRIPTION
**Pull request by rubenzorrilla**

### CODE REVIEW

File: applications/FluidDynamicsApplication/custom_processes/distance_modification_process.h [hunk 1]
```diff
@@ -207,80 +210,86 @@ class DistanceModificationProcess : public Process
...
-            mrModelPart.GetCommunicator().SynchronizeCurrentDataToMin(DISTANCE);
         }
         else // Case in where the original distance needs to be kept to track the interface (e.g. FSI)
         {
-            for (auto itNode=mrModelPart.NodesBegin(); itNode!=mrModelPart.NodesEnd(); itNode++)
+            const unsigned int NumThreads = OpenMPUtils::GetNumThreads();
+            std::vector<std::vector<unsigned int>> AuxModifiedDistancesIDs(NumThreads);
+            std::vector<std::vector<double>> AuxModifiedDistancesValues(NumThreads);
+
+            #pragma omp parallel shared(AuxModifiedDistancesIDs, AuxModifiedDistancesValues)
```
Comment (RiccardoRossi) [inline applications/FluidDynamicsApplication/custom_processes/distance_modification_process.h:234]: just to tell that the "shared" is automatic.

File: applications/FluidDynamicsApplication/custom_processes/distance_modification_process.h [hunk 2]
```diff
@@ -295,12 +304,22 @@ class DistanceModificationProcess : public Process
...
-            unsigned int nodeId = mModifiedDistancesIDs[i];
-            mrModelPart.GetNode(nodeId).FastGetSolutionStepValue(DISTANCE) = mModifiedDistancesValues[i];
+            const int ThreadId = OpenMPUtils::ThisThread();
+            const std::vector<unsigned int> LocalModifiedDistancesIDs = mModifiedDistancesIDs[ThreadId];
+            const std::vector<double> LocalModifiedDistancesValues = mModifiedDistancesValues[ThreadId];
+
+            for(unsigned int i=0; i<LocalModifiedDistancesIDs.size(); ++i)
+            {
+                const unsigned int nodeId = LocalModifiedDistancesIDs[i];
+                mrModelPart.GetNode(nodeId).FastGetSolutionStepValue(DISTANCE) = LocalModifiedDistancesValues[i];
```
Comment (RiccardoRossi) [inline applications/FluidDynamicsApplication/custom_processes/distance_modification_process.h:316]: you don't really have another option here, still i would like to make clear from a "didactic" point of view that the GetNode function has a cost that is "logN" where N is the number of nodes. 
this means that the cost of this loop is NlogN (not N as one might expect)

**Review by RiccardoRossi — APPROVED** [review]
just one "one million dollae question": do you have a small test for this?

i am speaking of something minimal, not going though the main...

File: applications/FluidDynamicsApplication/custom_processes/distance_modification_process.h (hunk 1 above)
Comment (rubenzorrilla) [reply applications/FluidDynamicsApplication/custom_processes/distance_modification_process.h:234]: I thought that "shared" clause was not automatic. Thanks!

File: applications/FluidDynamicsApplication/custom_processes/distance_modification_process.h (hunk 2 above)
Comment (rubenzorrilla) [reply applications/FluidDynamicsApplication/custom_processes/distance_modification_process.h:316]: Thanks for the info!!

**Comment by rubenzorrilla** [comment]
I compromise myself to do a small test ASAP.

**Pull request merged**